### PR TITLE
UI for entering pre-move survey information [delivers #159742476]

### PR DESCRIPTION
**Pull request by stangah**
## Description

Creates a shared Premove Survey editable panel that is used both by office and tsp apps.

## Setup
You'll need to have shipments/HHG moves in your database to be able to see the survey panel, add it (it also wipes your db) with

```sh
make db_populate_e2e
```

## Code Review Verification Steps

* [ ] End to end tests pass (`make e2e_test`).
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/blob/master/docs/backend.md#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/blob/master/docs/backend.md#querying-the-database-safely)
 have been satisfied.
* [ ] Any migrations/schema changes also update the diagram in docs/schema/dp3.sqs
* [ ] There are no aXe warnings for UI.
* [ ] This works in IE.
* Any new client dependencies (Google Analytics, hosted libraries, CDNs, etc) have been:
  * [ ] Communicated to @willowbl00
  * [ ] Added to the list of [network dependencies](https://github.com/transcom/mymove#client-network-dependencies)
* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/159742476) for this change

screenshot 2018-08-21 13 45 05
screenshot 2018-08-21 16 33 24
screenshot 2018-08-21 16 33 30

### CODE REVIEW

File: cypress/support/commands.js [hunk 1]
```diff
@@ -86,3 +87,81 @@ Cypress.Commands.add('upload_file', (selector, fileUrl) => {
       });
   });
 });
+
+Cypress.Commands.add('testPremoveSurvey', () => {
```
Comment (tinyels) [inline cypress/support/commands.js]: I'm not sure this needs to be a command. We could just put it in a separate file and import for the two places it is being called.

File: src/shared/EditablePanel/index.jsx [hunk 2]
```diff
@@ -36,10 +37,8 @@ export const SwaggerValue = props => {
   if (swaggerProps.format === 'cents') {
     value = formatCents(value);
   }
-  if (swaggerProps.type === 'integer') {
-    if (fieldName === 'weight_estimate') {
-      value = value.toLocaleString() + ' lbs';
-    }
+  if (swaggerProps['x-formatting'] === 'weight') {
```
Comment (tinyels) [inline src/shared/EditablePanel/index.jsx]: We will want to socialize this.

File: cypress/integration/office/officeUserShipment.js [hunk 3]
```diff
@@ -3,6 +3,9 @@ describe('office user finds the move', function() {
   beforeEach(() => {
     cy.signIntoOffice();
   });
+  it('office user enters premove survey', function() {
```
Comment (tinyels) [inline cypress/integration/office/officeUserShipment.js]: I would consider having this test in a separate file so it can be run independently from rest of hhg office tests. We haven't really come up with conventions about test structure but imho if a test is independent from others having it organized so it can be run by itself is handy. Also, it will be rare that an office user edits pre-move survey. (And we will probably prevent that in the future)

File: src/shared/PremoveSurvey/index.jsx [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+        </div>
+
+        <div className="editable-panel-3-column">
+          <SwaggerField
+            fieldName="pm_survey_weight_estimate"
+            swagger={schema}
+            className="half-width"
+            required
+          />
+          <SwaggerField
```
Comment (tinyels) [inline src/shared/PremoveSurvey/index.jsx]: The progear and spousal progear values need to allow for 0 or be optional.
image

File: cypress/integration/mymove/ppm.js [hunk 5]
```diff
@@ -71,7 +71,7 @@ describe('completing the ppm flow', function() {
     cy.nextPage();
     cy.contains('Success');
     cy.contains('Next Step: Awaiting approval');
-    cy.contains('Weight (est.): 6006 lbs');
+    cy.contains('Weight (est.): 6000 lbs');
```
Comment (tinyels) [inline cypress/integration/mymove/ppm.js]: Why did you change this?
image

Comment (tinyels) [reply cypress/integration/mymove/ppm.js]: Though we should probably fix the formatting here too--in another story.

Comment (stangah) [reply cypress/integration/mymove/ppm.js]: That's weird. I changed it because on my machine it was looking for 6000 and failing...

Comment (stangah) [reply cypress/integration/mymove/ppm.js]: So I think this is a byproduct of the `click` action on the slider handle, and depending maybe on what resolution you run the test at it might move it over a pixel to 6,006. It's not moving it on my machine when clicked for whatever reason. I'm going to remove the check on the weight and replace it with advance, which won't change.

File: pkg/testdatagen/scenario/e2ebasic.go [hunk 6]
```diff
@@ -242,4 +236,42 @@ func (e e2eBasicScenario) Run(db *pop.Connection, loader *uploader.Uploader) {
 	hhg0.Move.Submit()
 	// Save move and dependencies
 	models.SaveMoveDependencies(db, hhg0.Move)
+
+	// Service member with uploaded orders and an approved shipment
+	email = "hhg@award.ed"
+
+	offer1 := testdatagen.MakeShipmentOffer(db, testdatagen.Assertions{
+		User: models.User{
```
Comment (amitch23) [inline pkg/testdatagen/scenario/e2ebasic.go]: It doesn't look like this User is ever being created or associated to the Service Member, so the id and email here don't appear in the local login. A user is being created, but with a different UUID and email (first.last@login.gov.test). The same is true for the "hhg@incomple.te" user above (though I know you didn't add that, but I suspect there was a copy paste here).

Comment (amitch23) [reply pkg/testdatagen/scenario/e2ebasic.go]: (I mean, technically you can sign in, but you have to click on every first.last.@login... until you find kevin bacon.)

Comment (tinyels) [reply pkg/testdatagen/scenario/e2ebasic.go]: good catch @amitch23

**Comment by tinyels** [comment]
If it is not too much work, let's fix this too:
image
